data_owner_ids.py

```diff
@@ -18,15 +23,51 @@ def parse_args():
     return args
 
 
-def process_csv(csv_path, system_csv_path, system):
+def extract_metadata_and_zip(system, system_zip_path, system_output_dir_path, n_rows):
+    print(str(system_output_dir_path))
+    metadata_file_name = system_output_dir_path / (system + "_metadata.json")
+    with zipfile.ZipFile(system_zip_path) as system_zip:
+        metadata_files = []
+        for file_name in system_zip.namelist():
+            if "metadata" in file_name:
+                metadata_files.append(file_name)
+        if len(metadata_files) > 1:
+            warnings.warn(
+                f"Could not extract metadata from {system} - too many metadata files found in system archive"
+            )
+        elif len(metadata_files) < 1:
+            warnings.warn(
+                f"Could not extract metadata from {system} - no metadata file found in system archive"
+            )
+        else:
+            with system_zip.open(metadata_files[0], mode="r") as meta_fp:
+                metadata = json.load(meta_fp)
+            metadata["number_of_links"] = n_rows
+            with open(metadata_file_name, "w+") as fp:
+                json.dump(metadata, fp)
+    zpath = Path(str(system_output_dir_path) + ".zip", at=system + "/")
```

Comment:
I think the Linkage Agent will be using linux, but we should do our best to avoid specifying path separators in strings and use operators from Path.

Reply:
good catch - I thought I had changed all of those. will fix